lib/engine/game/g_18_zoo.rb

```
@@ -0,0 +1,128 @@
+                              'MONDAY'
+                            elsif index < 12
+                              'TUESDAY'
+                            elsif index < 16
+                              'WEDNESDAY'
+                            end
+            company
+          end.compact
+        end
+
+        def priority_deal_player
+          return @players.first if @round.is_a?(Round::Draft)
+
+          super
+        end
+
+        # Only buy and sell par shares is possible action during SR
+        def stock_round
+          Round::Stock.new(self, [
+              Step::BuySellParShares,
+          ])
+        end
+
+        def game_end_check
+          triggers = {
+              custom: custom_end_game_reached?,
+          }.select { |_, t| t }
+
+          %i[immediate current_round current_or full_or one_more_full_or_set].each do |after|
+            triggers.keys.each do |reason|
+              if game_end_check_values[reason] == after
+                (@turn == (@final_turn ||= @turn + 1)) if after == :one_more_full_or_set
+                return [reason, after]
+              end
+            end
+          end
+
+          nil
+        end
+      end
+    end
+
+    class G18ZOOMapA < Internal::G18ZOO
+      load_from_json(Config::Game::G18ZOO::JSON, Config::Game::G18ZOOMapA::JSON)
+
+      def self.title
+        '18ZOO - Map A'
+      end
+    end
+
+    class G18ZOOMapB < G18ZOOMapA
+      load_from_json(Config::Game::G18ZOO::JSON, Config::Game::G18ZOOMapB::JSON)
+
+      def self.title
+        '18ZOO - Map B'
+      end
+    end
+
+    class G18ZOOMapC < G18ZOOMapA
+      load_from_json(Config::Game::G18ZOO::JSON, Config::Game::G18ZOOMapC::JSON)
+
+      def self.title
+        '18ZOO - Map C'
+      end
+    end
+
+    class G18ZOOMapD < Internal::G18ZOO
+      load_from_json(Config::Game::G18ZOO::JSON, Config::Game::G18ZOOMapD::JSON)
+
+      def self.title
+        '18ZOO - Map D'
+      end
+    end
+
+    class G18ZOOMapE < G18ZOOMapD
```

you have a different game for every map? can you instead leverage the optional rules system we have now?

---

the optional rule systems uses checkbox (so you can select multiple); how can I be sure user choose just one? should I throw an error if the creation is done without any map, or with more than one?

---

you can just throw an error for now